pennylane/templates/state_preparations/cosine_window.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the CosineWindow template.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.operation import AnyWires, StatePrepBase
+from pennylane import math
+from pennylane.wires import Wires, WireError
+
+
+class CosineWindow(StatePrepBase):
+    r"""CosineWindow(wires)
+    Prepares an initial state with a cosine wave function.
+
+    The wave function is defined as
+
+    .. math::
+
+        |\psi\rangle = \sqrt{2^{1-m}} \sum_{k=0}^{2^m-1} \cos(\frac{\pi k}{2^m} - \frac{\pi}{2}) |k\rangle,
+
+    where :math:`m` is the number of wires.
+
+    .. figure:: ../../_static/templates/state_preparations/cosine_window.png
+        :align: center
+        :width: 65%
+        :target: javascript:void(0);
+
+    .. note::
+
+        The wave function is shifted by :math:`\frac{\pi}{2}` units so that the window is centered.
+
+
+    Args:
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+        id (str): custom label given to an operator instance,
+            can be useful for some applications where the instance has to be identified.
+
+    **Example**
+
+    >>> dev = qml.device('default.qubit', wires=2)
+    >>> @qml.qnode(dev)
+    ... def example_circuit():
+    ...     qml.CosineWindow(wires=range(2))
+    ...     return qml.probs()
+    >>> print(example_circuit())
+    [1.87469973e-33 2.50000000e-01 5.00000000e-01 2.50000000e-01]
+    """
+    num_wires = AnyWires
+    num_params = 0
```

Comment:
Just fyi, no need to change: 
- The default for num_wires is `AnyWires` when its not specified
- The default value for num_params is the number of non-keyword arguments passed into the init of the class.